packages/html/src/events/visible.rs

```diff
@@ -0,0 +1,325 @@
+use std::fmt::{Display, Formatter};
+
+pub struct VisibleData {
+    inner: Box<dyn HasVisibleData>,
+}
+
+impl<E: HasVisibleData> From<E> for VisibleData {
+    fn from(e: E) -> Self {
+        Self { inner: Box::new(e) }
+    }
+}
+
+impl VisibleData {
+    /// Create a new VisibleData
+    pub fn new(inner: impl HasVisibleData + 'static) -> Self {
+        Self {
+            inner: Box::new(inner),
+        }
+    }
+
+    /// Get the bounds rectangle of the target element
+    pub fn get_bounding_client_rect(&self) -> VisibleResult<PixelsRect> {
+        self.inner.get_bounding_client_rect()
+    }
+
+    /// Get the ratio of the intersectionRect to the boundingClientRect
+    pub fn get_intersection_ratio(&self) -> VisibleResult<f64> {
+        self.inner.get_intersection_ratio()
+    }
+
+    /// Get the rect representing the target's visible area
+    pub fn get_intersection_rect(&self) -> VisibleResult<PixelsRect> {
+        self.inner.get_intersection_rect()
+    }
+
+    /// Get if the target element intersects with the intersection observer's root
+    pub fn is_intersecting(&self) -> VisibleResult<bool> {
+        self.inner.is_intersecting()
+    }
+
+    /// Get the rect for the intersection observer's root
+    pub fn get_root_bounds(&self) -> VisibleResult<PixelsRect> {
+        self.inner.get_root_bounds()
+    }
+
+    /// Get the element whose intersection with the root changed
+    pub fn get_target(&self) -> VisibleResult<ElementId> {
+        self.inner.get_target()
+    }
+
+    /// Get a timestamp indicating the time at which the intersection was recorded
+    pub fn get_time(&self) -> VisibleResult<f64> {
```

Comment:
Could we return an instant here instead of a float? I'm guessing seconds, but it isn't clear what the unit is looking at the function signature.